crates/ra_hir_def/src/body/lower.rs

```diff
@@ -547,6 +557,21 @@ impl ExprCollector<'_> {
         }
     }
 
+    fn find_inner_item<S: ItemTreeSource>(&self, id: AstId<ast::ModuleItem>) -> FileItemTreeId<S> {
+        let index =
+            self.item_trees.iter().position(|(file, _)| *file == id.file_id).unwrap_or_else(|| {
```

Comment:
This feels like it might turn into "oups, acidentally quadratic" bug after several months. But that's probably ok. Perhaps just binary_search it?